tests-arquillian/src/test/java/org/jboss/weld/tests/jsf/flow/FlowMultipleWarTest.java

```diff
@@ -0,0 +1,117 @@
+import org.jboss.weld.tests.category.Integration;
+import org.jboss.weld.tests.util.PropertiesBuilder;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import com.gargoylesoftware.htmlunit.FailingHttpStatusCodeException;
+import com.gargoylesoftware.htmlunit.WebClient;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import com.gargoylesoftware.htmlunit.html.HtmlSubmitInput;
+
+/**
+ * See also WELD-2448. If multiple deployments share a proxy class from static module (JSF impl in this test case) and each deployment uses a different value
+ * for {@link ConfigurationKey#RELAXED_CONSTRUCTION} the proxy instance might not be initialized corretly.
+ *
+ * @author Martin Kouba
+ */
+@Category(Integration.class)
+@RunWith(Arquillian.class)
+public class FlowMultipleWarTest {
+
+    private static final String WAR1 = "war1";
+
+    private static final String WAR2 = "war2";
+
+    @Deployment(name = WAR1, managed = false, testable = false)
+    public static WebArchive createFooTestArchive() {
+        WebArchive war1 = ShrinkWrap.create(WebArchive.class, WAR1 + ".war")
+                .addAsWebResource(FlowMultipleWarTest.class.getPackage(), "index.xhtml",
+                        "index.xhtml")
+                .addAsWebResource(FlowMultipleWarTest.class.getPackage(), "simple-flow-flow.xml",
+                        "simple-flow/simple-flow-flow.xml")
+                .addAsWebResource(FlowMultipleWarTest.class.getPackage(), "simple-flow.xhtml",
+                        "simple-flow/simple-flow.xhtml")
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml").addClass(Foo.class)
+                .setWebXML(FlowMultipleWarTest.class.getPackage(), "web.xml");
+        return war1;
+    }
+
+    @Deployment(name = WAR2, managed = false, testable = false)
+    public static WebArchive createBarTestArchive() {
+        return ShrinkWrap.create(WebArchive.class, WAR2 + ".war")
+                .addAsWebResource(FlowMultipleWarTest.class.getPackage(), "index.xhtml",
+                        "index.xhtml")
+                .addAsWebResource(FlowMultipleWarTest.class.getPackage(), "simple-flow-flow.xml",
+                        "simple-flow/simple-flow-flow.xml")
+                .addAsWebResource(FlowMultipleWarTest.class.getPackage(), "simple-flow.xhtml",
+                        "simple-flow/simple-flow.xhtml")
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml").addClass(Bar.class)
+                .setWebXML(FlowMultipleWarTest.class.getPackage(), "web.xml").addAsResource(
+                        PropertiesBuilder.newBuilder()
+                                .set(ConfigurationKey.RELAXED_CONSTRUCTION.get(), "true").build(),
+                        "weld.properties");
+    }
+
+    @ArquillianResource
+    Deployer deployer;
+
+    /**
+     * This is not a real test method - we just need to deploy WAR1 first.
```

Comment:
I think you could just use `@Before` method.

Reply:
Probably, but only if you declare a single test method. This approach is imho more robust.

Reply:
`@Before` can be bound to each method, so it would work woth multiple methods as well.
BTW you might also want to call `deployer.undeploy()`. I think Arq. does it on its own, but it looks cleaner to handle it.

Reply:
> @Before can be bound to each method, so it would work woth multiple methods as well.

Sure, but we only want to perform the deployment once.

> `deployer.undeploy()`

Good point.

Reply:
I just wanted to avoid having this counted as a test method when it's actually a setup.

Reply:
I see. Actually, we can merge those two methods into one I think...

Reply:
Hm, probably not as `@ArquillianResource` would not work.

Reply:
You would need to know the URL yourself then, or extract it from somewhere.